exporter/loadbalancingexporter/resolver_dns.go

```diff
@@ -0,0 +1,171 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loadbalancingexporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"sort"
+	"sync"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"go.uber.org/zap"
+)
+
+var _ resolver = (*dnsResolver)(nil)
+
+const (
+	defaultResInterval = 5 * time.Second
+	defaultResTimeout  = time.Second
```

Comment:
Is this reasonably enough time to resolve a hostname? I'd imagine it is quite possible to take more time.

Reply:
In the common case, 100ms is already considered a high latency: either the authoritative server is close enough (like in Kubernetes), or caches are expected to exist. But even if it fails once or twice in a row, it's typically not a problem, as changes aren't happening every second.